components/script/dom/audiobuffer.rs

```diff
@@ -262,7 +262,7 @@ impl AudioBufferMethods for AudioBuffer {
         let cx = GlobalScope::get_cx();
         let channel_number = channel_number as usize;
         let offset = start_in_channel as usize;
-        let mut dest = Vec::with_capacity(destination.len());
+        let mut dest = vec![0.0_f32; bytes_to_copy];
```

Comment:
Note: the previous code was using `extend_from_slice` on the vec, while the new code is using `copy_from_slice` on the underlying slice, hence the crash.